force-app/main/default/classes/CQ_KR_Test_PlayerHandler.cls

```diff
@@ -0,0 +1,88 @@
+@isTest
+public class CQ_KR_Test_PlayerHandler {
+
+    // Helper method to create a user with a specific permission set
+    private static User createUserWithPermissionSet() {
+        // Retrieve the desired permission set by name
+        PermissionSet ps = [SELECT Id FROM PermissionSet WHERE Name = 'CQ_KR_Common_Permission_Set' LIMIT 1];
+        
+        // Retrieve a profile for the user (replace 'Standard User' with the actual profile name)
+        Profile p = [SELECT Id FROM Profile WHERE Name = 'Standard User' LIMIT 1]; 
+        
+        // Create a new user with specified attributes
+        User u = new User(
+            ProfileId = p.Id,
+            Alias = 'standt',
+            Email = 'standarduser@testorg.com',
+            EmailEncodingKey = 'UTF-8',
+            LastName = 'Testing',
+            LanguageLocaleKey = 'en_US',
+            LocaleSidKey = 'en_US',
+            TimeZoneSidKey = 'America/Los_Angeles',
+            UserName = 'standarduser@testo1rg.com'
+        );
+        // Insert the user and assign the permission set
+        insert u;
+        PermissionSetAssignment psa = new PermissionSetAssignment(
+            AssigneeId = u.Id,
+            PermissionSetId = ps.Id
+        );
+        insert psa;
+        
+        return u;  // Return the created user
+    }
+    
+    @isTest
+    static void testBeforeUpdate() {
+        // Run the test as the user with the specified permission set
+        System.runAs(createUserWithPermissionSet()) {
```

Comment:
we dont need to create user in every methods

Reply:
ok sir